pkg/antctl/raw/test/client.go

```diff
@@ -0,0 +1,188 @@
+	return c.Clientset.AppsV1().Deployments(namespace).Delete(ctx, name, opts)
+}
+
+func (c *Client) DeploymentIsReady(ctx context.Context, namespace, deployment string) error {
+	d, err := c.GetDeployment(ctx, namespace, deployment, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	if d == nil {
+		return fmt.Errorf("deployment is not available")
+	}
+
+	if d.Status.Replicas == 0 {
+		return fmt.Errorf("replicas count is zero")
+	}
+
+	if d.Status.AvailableReplicas != d.Status.Replicas {
+		return fmt.Errorf("only %d of %d replicas are available", d.Status.AvailableReplicas, d.Status.Replicas)
+	}
+
+	if d.Status.ReadyReplicas != d.Status.Replicas {
+		return fmt.Errorf("only %d of %d replicas are ready", d.Status.ReadyReplicas, d.Status.Replicas)
+	}
+
+	if d.Status.UpdatedReplicas != d.Status.Replicas {
+		return fmt.Errorf("only %d of %d replicas are up-to-date", d.Status.UpdatedReplicas, d.Status.Replicas)
+	}
+
+	return nil
+}
+
+func (c *Client) CreateNamespace(ctx context.Context, namespace string, opts metav1.CreateOptions) (*corev1.Namespace, error) {
+	return c.Clientset.CoreV1().Namespaces().Create(ctx, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}}, opts)
+}
+
+func (c *Client) GetNamespace(ctx context.Context, namespace string, options metav1.GetOptions) (*corev1.Namespace, error) {
+	return c.Clientset.CoreV1().Namespaces().Get(ctx, namespace, options)
+}
+
+func (c *Client) DeleteNamespace(ctx context.Context, namespace string, opts metav1.DeleteOptions) error {
+	return c.Clientset.CoreV1().Namespaces().Delete(ctx, namespace, opts)
+}
+
+func (c *Client) DeletePod(ctx context.Context, namespace, name string, opts metav1.DeleteOptions) error {
+	return c.Clientset.CoreV1().Pods(namespace).Delete(ctx, name, opts)
+}
+
+func (c *Client) ListPods(ctx context.Context, namespace string, options metav1.ListOptions) (*corev1.PodList, error) {
+	return c.Clientset.CoreV1().Pods(namespace).List(ctx, options)
+}
+
+func (c *Client) ExecInPod(ctx context.Context, namespace, pod, container string, command []string) (bytes.Buffer, error) {
+	result, err := c.execInPod(ctx, ExecParameters{
+		Namespace: namespace,
+		Pod:       pod,
+		Container: container,
+		Command:   command,
+	})
+	if err != nil {
+		return bytes.Buffer{}, err
+	}
+
+	if errString := result.Stderr.String(); errString != "" {
+		return bytes.Buffer{}, fmt.Errorf("command failed: %s", errString)
+	}
+
+	return result.Stdout, nil
+}
+
+func (c *Client) GetDaemonSet(ctx context.Context, namespace, name string, opts metav1.GetOptions) (*appsv1.DaemonSet, error) {
+	return c.Clientset.AppsV1().DaemonSets(namespace).Get(ctx, name, opts)
+}
+
+func (c *Client) DeleteDaemonSet(ctx context.Context, namespace, name string, opts metav1.DeleteOptions) error {
+	return c.Clientset.AppsV1().DaemonSets(namespace).Delete(ctx, name, opts)
+}
+
+type Kind int
```

Comment:
what is this for?

Reply:
that was for future implementation to describe the kind cluster. I will remove it for now to avoid confusion